java/vector/src/main/java/org/apache/arrow/vector/dictionary/DictionaryHashTable.java

```diff
@@ -135,7 +139,8 @@ public int getIndex(int indexInArray, ValueVector toEncode) {
     for (DictionaryHashTable.Entry e = table[index]; e != null ; e = e.next) {
       if ((e.hash == hash)) {
         int dictIndex = e.index;
-        if (dictionary.equals(dictIndex, toEncode, indexInArray)) {
+        visitor.reset(indexInArray, dictIndex, 1);
```

Comment:
This saves the repeated allocations. Can we also save the repeated type checks ?

Reply:
Thanks for you careful check, right, here checks type for each value compare.
The dictionary vector is bind with hashTable but the toEncode vector is not(that is to see a same dictionary could be used for multiple toEncode vectors), so it's hard to check which is the first value of toEncode to compare. But as a workaround, I would like to disable all type check here as I think they will always have same type, or we could make a check outside hash table. What do you think?